scanners/third_parties.py

```diff
@@ -125,146 +63,34 @@ def init_domain(domain, environment, options):
-        serialize(services['internal'])
-    ] + known_matches]
-
-
-# Given a 'domains' array from phantomas output, create a data
-# object with data on detected external, internal, and
-# affiliated domains.
-def services_for(url, data, domain, options):
-    services = {
-        'url': url,
-        'external': [],
-        'internal': [],
-        'known': [],
-    }
-
-    # break up data into a map of host to number of requests
-    hosts = {}
-    for string in data['offenders']['domains']:
-        pieces = string.split(": ")
-        hostname = pieces[0].lower()  # lowercase to be sure
-        number = int(pieces[1].split(" ")[0])
-        hosts[hostname] = number
-
-    # make iteration consistent
-    hostnames = list(hosts.keys())
-    hostnames.sort()
-
-    for host in hostnames:
-        if host.startswith("www."):
-            www_host = host
-            base_host = re.sub("^www.", "", host)
-        else:
-            www_host = "www.%s" % host
-            base_host = host
-
-        www_domain = "www.%s" % domain
-
-        # skip if it's the same, or is www version
-        if (
-            (base_host == domain) or (www_host == domain) or
-            (base_host == www_domain) or (www_host == www_domain)
-        ):
-            continue
-
-        # internal if it shares the same base domain
-        if (utils.base_domain_for(host) == utils.base_domain_for(domain)):
-            services['internal'].append(host)
-        # otherwise external
-        else:
-            services['external'].append(host)
-
-        # compare this host to all known services
-        for service in known_services:
-            for pattern in known_services[service]:
-
-                # exact string match
-                if isinstance(pattern, str):
-                    if host == pattern:
-                        services['known'].append(service)
-
-                # regular expression match
-                else:
-                    if re.search(pattern, host):
-                        services['known'].append(service)
-
-    # For each category, count up the requests
-    categories = list(services.keys())
-    categories.sort()
-    for category in ['external', 'internal']:
-        total = 0
-        for host in services[category]:
-            total += hosts[host]
-        services["%s_requests" % category] = total
-
-    return services
-
-
-def clean_domain_output(output):
-    return output.split(" ")[0]
-
-
-# Jam multiple domains into one CSV cell.
-def serialize(domains):
-    return str.join(', ', domains)
-
-
-base_fields = [
+        data['url'],
+        len(data['external_domains']),
+        str.join(" | ", data['external_domains']),
+        str.join(" | ", data['external_urls']),
+        str.join(" | ", data['nearby_domains']),
+        str.join(" | ", data['nearby_urls']),
+        str.join(" | ", data['known_services']),
+        str.join(" | ", data['unknown_services'])
+    ]]
```

Comment:
The URLs and domains should all be URL-encoded, so ` | ` should be a tolerable delimiter.